a10_neutron_lbaas/acos/openstack_mappings.py

```diff
@@ -99,10 +99,13 @@ def vip_protocols(c, os_protocol):
 def vport_swap_protocols(c, os_protocol):
     z = c.client.slb.virtual_server.vport
     # what it is: what it needs to be
+
     mapping = {
         z.TCP: z.HTTPS,
         z.HTTPS: z.TCP,
         z.HTTP: z.TCP
     }
 
-    return mapping[os_protocol]
+    # return if exists, else what we passed.
+    # damn unicode.
+    return mapping.get(str.lower(str(os_protocol))) or str(os_protocol)
```

Comment:
This is really suspicious.

I doubt this function's needed at all. You should always now what protocol you want.